project-6/project_6/main.py

```diff
@@ -0,0 +1,198 @@
+
+
+# load data into dataframe
+def load_data_from_file(path: str) -> pd.DataFrame:
+    """
+    Loads data from parquet file into pandas dataframe
+
+    Parameters
+    ----------
+        path : str
+            filepath to the parquet file
+
+    Returns
+    -------
+        Pandas DataFrame
+    """
+    try:
+        data = pd.read_parquet(path)
+    except:
+        raise SyntaxError("unable to load data from path")
+
+    logging.info("Data loaded into dataframe successfully")
+
+    return data
+
+
+# data validation
+def type_check(row: dict) -> bool:
+    """
+    Validates data by checking the data type.
+    If data type matches, then it returns True else False
+
+    Parameters
+    ----------
+        row : dict
+            variable containing values of a row
+
+    Returns
+    -------
+        Boolean
+    """
+
+    # dictionary containing data types that will be used for validation
+    dict_row_type = {
+        "ConstituentID": int,
+        "DisplayName": str,
+        "Nationality": str,
+        "Gender": str,
+        "BeginDate": int,
+        "EndDate": int,
+        "Wiki QID": str,
+        "ULAN": float,
+        "Title": str,
+        "Medium": str,
+        "Dimensions": str,
+        "CreditLine": str,
+        "AccessionNumber": str,
+        "Classification": str,
+        "Department": str,
+        "DateAcquired": datetime.datetime,
+        "Cataloged": str,
+        "ObjectID": float,
+        "URL": str,
+        "ThumbnailURL": str,
+        "Height (cm)": float,
+        "Width (cm)": float,
+        "completedDate": float,
+        "DateAcquired_year": int,
+        "DateAcquired_month": int,
+        "DateAcquired_day": int,
+        "DateAcquired_weekday": int,
+    }
+
+    # check for datatypes of input against the validation dictionary
+    for idx in row.keys():
+        if isinstance(row[idx], dict_row_type[idx]) is False:
+            logging.error(f"type check failed for :{idx}, {row[idx]}")
+            return False
+    return True
+
+
+# ingest data
+def ingest_data(data: pd.DataFrame) -> None:
+    """
+    Ingest data into postgres database from pandas dataframe
+
+    Parameters
+    ----------
+        data : pd.DataFrame
+            data to be ingested into dataframe
+
+    Returns
+    -------
+        None
+    """
+
+    # iterate through the data to insert the values into the database
+    for idx in data.iterrows():
+        try:
+            row = idx[1].to_dict()
+
+            logging.info("data ingestion into database started")
+            # validate the row before inserting into db
+            if type_check(row):
+                line = Artwork()
+                for key, value in row.items():
+                    setattr(line, key, value)
+                session.add(line)
+                session.commit()
+            else:
+                logging.error(f"type check failed for {idx[0]}:{row}")
```

Comment:
upper case for first letter - apply to all logs

Reply:
All Done